zeppelin-interpreter/src/test/java/org/apache/zeppelin/completer/LSPUtilsTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.completer;
+
+import org.apache.zeppelin.interpreter.thrift.InterpreterCompletion;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.List;
+
+import static org.eclipse.lsp4j.jsonrpc.Launcher.createLauncher;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+public class LSPUtilsTest{
+  private static final String HOST = "localhost";
+  private static final String BUF = "p";
+  private static final String LANG_ID = "python";
+  private static final int CURSOR = 0;
+
+  @Test
+  public void getLspServerComplitionsTest() throws Exception{
+
+    try (ServerSocket serverSocket = new ServerSocket(0)) {
+      final int port = serverSocket.getLocalPort();
+
+      Thread t = new Thread(() -> {
```

Comment:
Making thread has timing issue sometimes. How about using `@Mock`?

Reply:
`@Mock` does not work with static methods. For this, `powermock` library is needed which will be additional dependency.

Reply:
Even if we should `powermock`, it looks better than making tests fragile, in my opinion.

Reply:
You are right, fragile test is bad practice. I changed it to PowerMock.